Adds support for OpenCLIP

LysandreJik opened the pull request:
Adds support for OpenCLIP as a library with a code snippet.

cc @rwightman

osanseviero commented:
(this is cool!)

For usage of `hf_hub_download` in openclip, setting `library_name` and `library_version` is also nice :)

LysandreJik commented:
Thanks for the feedback, I'll add that to the OpenCLIP PR!

rwightman commented:
Looks good, 

@osanseviero you mean? https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/pretrained.py#L14

osanseviero commented:
Yes, maybe https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/pretrained.py#L282 should also change but I'm not very familiar with the codebase

rwightman commented:
> Yes, maybe https://github.com/mlfoundations/open_clip/blob/main/src/open_clip/pretrained.py#L282 should also change (which is the function that made me say the comment above). but I'm not very familiar with the codebase

It's using the fn I linked to above, it bound the lib/version args

julien-c commented:
i don't think `library_name` and `library_version` are used in Elasticsearch stats currently (unless i'm mistaken?)

osanseviero commented:
Yes, this does not impact download stats but this should allow us to do fine grained analysis at any point in the future if there's any interest in that. We have this in all the integrations.

julien-c commented:
yes i was only commenting on @rwightman's question 
> if there are a decent number of existing downloads [based on this]

rwightman commented:
we should probably line up the names across lib name and tags either way? but good to know re elasticsearch stats.

LysandreJik commented:
Updated the code snippet to also feature the tokenizer and updated `openclip` to `open_clip` to be coherent.

Switching this to non-draft mode as https://github.com/mlfoundations/open_clip/pull/375 was merged.

Pierrci commented on js/src/lib/interfaces/Libraries.ts:
```diff
@@ -16,6 +16,7 @@ export enum ModelLibrary {
 	"keras"                  = "Keras",
 	"k2"                     = "K2",
 	"nemo"                   = "NeMo",
+	"open_clip"               = "OpenCLIP",
```
```suggestion
	"open_clip"              = "OpenCLIP",
```
😃

LysandreJik replied:
oops, good catch! Thanks @Pierrci

Pierrci reviewed (commented):
LGTM

LysandreJik commented:
Thanks for the reviews @Pierrci and @coyotte508!
